### PR TITLE
Disable login

### PR DESCRIPTION
**Pull request by matthewmoto**
I needed this on my local instance and it may address issue #297 so I figured I'd share to see if you want it. Some of the folks in that thread are talking about supporting remote authentication (as one would using LDAP, etc), but this patch allows simply disabling login and pretends any user is an authenticated "dummy" user with an ID of -1 (regular users start at 1,2,3...) for any audit entries in the DB (currently only django_admin_log as far as I can tell). This also allows people to turn this on/off at will without having entries from the dummy user show up as a real one (which might be important in the future).

### CODE REVIEW

File: src/paperless/auto_auth.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth.models import User
```
Comment (danielquinn) [inline src/paperless/auto_auth.py]: I'm not sure why you're importing `User` here if you're going to redefine it below.

File: src/paperless/settings.py [hunk 2]
```diff
@@ -77,14 +77,23 @@
 if os.getenv("PAPERLESS_INSTALLED_APPS"):
     INSTALLED_APPS += os.getenv("PAPERLESS_INSTALLED_APPS").split(",")
 
+
+#Default Django authentication middleware (requires a username/password)
```
[the remark this answers is not in the document]
Comment (matthewmoto) [reply src/paperless/settings.py]: Thanks for taking the time to look at the changes and suggest improvements. I'll merge them in and clean things up or another round in the next few days.

**Comment by matthewmoto** [comment]
I've moved the code around and cleaned it up according the the change requests. Feel free to ping me if I missed or misunderstood anything.

**Comment by danielquinn** [comment]
Sorry this has taken so long to get around to, but I've got 2 other projects I'm juggling on top of a day job :-)  Merging now!

**Pull request merged**